sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/DiscountServiceImpl.java

```diff
@@ -26,6 +28,18 @@ public Discount getById(Long id) {
         return discountDAO.findAll();
     }
 
+    @Override
+    public List<Discount> getByNameOrDescriptionContaining(String searchText) {
+        List<Discount> result = new ArrayList<>();
+        if (searchText != null && !searchText.isEmpty()) {
+            result = discountDAO.findByNameContainingOrDescriptionContaining(searchText);
+        }
+        if (result.isEmpty()) {
```

Comment:
Why do we need this? Seems we should return empty resp if search returns no data

Reply:
I asked , what this method will return?
Vladimir sad "Please return all items if searchWork is empty. Also please don't forget to remove this comment"

Reply:
We have search request on page where we see all discounts and when user use search without criteria, we refresh all discounts

Reply:
At this case case use smth like 
```
        if (searchText != null &amp;&amp; !searchText.isEmpty()) {           
                result = discountDAO.findByNameContainingOrDescriptionContaining(searchText);        
        }  else {
                 result = discountDAO.findAll();
        }
```

With current logic we will return irrelevant result